### PR TITLE
chore: introduce o-colors to private foundation

### PR DESCRIPTION
**Pull request by frshwtr**
## Describe your changes
This is still WIP. Raising to visually see changes

## Checklist before requesting a review

- [ ] I have applied `percy` label for o-[COMPONENT] or `chromatic` label for o3-[COMPONENT] on my PR before merging and after review. Find more details in [CONTRIBUTING.md](https://github.com/Financial-Times/origami/blob/main/CONTRIBUTING.md#pull-requests-and-visual-regression-tests)
- [ ] If it is a new feature, I have added thorough tests.
- [ ] I have updated relevant docs.
- [ ] I have updated relevant env variables in Doppler.

### CODE REVIEW

File: components/o-forms/main.scss [hunk 1]
```diff
@@ -1,7 +1,6 @@
 @import '@financial-times/math';
 @import '@financial-times/o-spacing/main';
 @import '@financial-times/o-brand/main';
-@import '@financial-times/o-private-foundation/src/scss/o-grid/main';
```
Comment (frshwtr) [inline components/o-forms/main.scss:4]: Spacing should be removed here

File: components/o-header-services/src/scss/_base.scss [hunk 2]
```diff
@@ -1,3 +1,4 @@
+@import '@financial-times/o-private-foundation/o-normalise/main';
```
Comment (frshwtr) [inline components/o-header-services/src/scss/_base.scss:1]: Remove

File: components/o-private-foundation/o-normalise/main.scss [hunk 3]
```diff
@@ -0,0 +1,2 @@
+@import './src/scss/variables';
```
Comment (frshwtr) [inline components/o-private-foundation/o-normalise/main.scss:1]: ??

File: components/o-private-foundation/o-normalise/src/scss/_mixins.scss [hunk 4]
```diff
@@ -0,0 +1,126 @@
+@import '@financial-times/o-colors/main';
```
Comment (frshwtr) [inline components/o-private-foundation/o-normalise/src/scss/_mixins.scss:1]: why is this here?

File: components/o-subs-card/src/scss/_mixins.scss [hunk 5]
```diff
@@ -1,3 +1,4 @@
+@import '@financial-times/o-private-foundation/o-normalise/main';
```
Comment (frshwtr) [inline components/o-subs-card/src/scss/_mixins.scss:1]: remove

File: components/o-teaser/src/scss/_mixins.scss [hunk 6]
```diff
@@ -1,3 +1,4 @@
+@import '@financial-times/o-private-foundation/o-normalise/main';
```
Comment (frshwtr) [inline components/o-teaser/src/scss/_mixins.scss:1]: remove

File: components/o-private-foundation/src/scss/o-colors/_palette.scss [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+		('o3-color-palette-light-blue', #00a0dd),
+		('o3-color-palette-crimson', #cc0000),
+		('o3-color-palette-graphics-dark-blue', #006f9b),
+		('o3-color-palette-wheat', #f2dfce, ('allow-tones': true)),
+		('o3-color-palette-candy', #ff7faa),
+		('o3-color-palette-wasabi', #96cc28),
+		('o3-color-palette-jade', #00994d),
+		('o3-color-palette-velvet', #593380),
+		('o3-color-palette-lemon', #ffec1a),
+		('o3-color-palette-sky', #cce6ff),
```
Comment (notlee) [inline components/o-private-foundation/src/scss/o-colors/_palette.scss:36]: Aren't most of these coming from o3 tokens now?

File: components/o-private-foundation/src/scss/o-colors/_variables.scss [hunk 8]
```diff
@@ -0,0 +1,70 @@
+/// Ensure default mixins are only called once
+/// @type Bool
+/// @access private
+$_o-pf-colors-is-initialised: false !default;
```
Comment (notlee) [inline components/o-private-foundation/src/scss/o-colors/_variables.scss:4]: Can we delete all (or some?) of these empty private variables?

File: components/o-forms/src/scss/shared/_brand.scss [hunk 9]
```diff
@@ -11,61 +11,61 @@
...
-		controls-base: oColorsByName('teal'),
-		controls-negative-checked-background: oColorsByName('teal-30'),
-		toggle-inverse-selected: oColorsByName('teal-100'),
-		invalid-base: oColorsByName('crimson'),
-		invalid-base-inverse: oColorsByName('white'),
-		invalid-base-color-inverse: oColorsByName('black'),
-		valid-base: oColorsMix('jade', 'black', 80),
-		toggle-inverse-knob: oColorsByName('slate'),
+		toggle-selected: oPrivateFoundationGet('o3-color-palette-teal-50'),
+		controls-base: oPrivateFoundationGet('o3-color-palette-teal-50'),
```
[the remark this answers is not in the document]
Comment (notlee) [reply components/o-forms/src/scss/shared/_brand.scss:33]: Hex is the same so it shouldn't matter, but we could in theory decide we want different tints. It's not not unlikely. Currently they're based on changing the B of HSB (or the L or HSL, I forget) which is pretty old school and can lead to muddy colours.

File: components/o-private-foundation/src/scss/o-colors/_functions.scss [hunk 10]
```diff
@@ -0,0 +1,422 @@
...
+/// colour name and its tone brightness, otherwise return null.
+///
+/// @example Get the tone brightness of 'claret-80'
+///    $tone-details: oPrivateColorsGetToneDetails('claret-80');
+///    $color-name: map-get($tone-details, 'color-name'); // claret
+///    $brightness: map-get($tone-details, 'brightness'); // 80
+///
+/// @param {String} $color - the palette colour or color name e.g. 'claret-80'
+/// @return {Map|Null} - the details of the given tone e.g. ('color-name': 'claret', 'brightness': 80)					)
+@function oPrivateColorsGetToneDetails($color) {
```
Comment (notlee) [inline components/o-private-foundation/src/scss/o-colors/_functions.scss:124]: This is only used by o-buttons. We don't need to upgrade o-buttons as we're dropping it. So we can revert the changes to o-buttons and delete this

File: components/o-private-foundation/src/scss/o-colors/_functions.scss [hunk 11]
```diff
@@ -0,0 +1,422 @@
...
+/// Returns a brighter or darker tone of a colour, where the hue remains
+/// the same but the saturation and luminance changes.
+///
+/// Not all our colours allow tones. If a colour cannot be toned an error is
+/// thrown. You may however mix the colour with another supported colour.
+/// @see oPrivateColorsMix
+///
+/// @param {String} $color-name - the name of the color to be shaded
+/// @param {Number} $brightness - the brightness value of the new color, 0-100
+@function oPrivateColorsGetTone($color-name, $brightness) {
```
Comment (notlee) [inline components/o-private-foundation/src/scss/o-colors/_functions.scss:75]: I think we have all tones aka tints within o3? So hopefully there's a route to delete this

File: components/o-private-foundation/src/scss/o-colors/_functions.scss (hunk 10 above)
Comment (frshwtr) [reply components/o-private-foundation/src/scss/o-colors/_functions.scss:124]: Will it still be needed for `private-foundation/o-buttons`? We have some [usages here](https://github.com/Financial-Times/origami/blob/6bf39a0b5f14fdf82730464f638dd0a5c0e6cf90/components/o-private-foundation/src/scss/o-buttons/custom-themes.scss#L606)

Comment (notlee) [reply components/o-private-foundation/src/scss/o-colors/_functions.scss:124]: Yes, because that will be used going forward

File: components/o-private-foundation/src/scss/o-colors/_functions.scss (hunk 11 above)
Comment (frshwtr) [reply components/o-private-foundation/src/scss/o-colors/_functions.scss:75]: There is, though it looks like a large refactor.

`oPrivateColorsGetTone` is buried deep in the logic to generate a [color map for buttons](https://github.com/Financial-Times/origami/blob/6bf39a0b5f14fdf82730464f638dd0a5c0e6cf90/components/o-private-foundation/src/scss/o-buttons/custom-themes.scss#L58-L90). This gives the [theme mixin context on what colours to set](https://github.com/Financial-Times/origami/blob/7a5ae1026fe421fd2ff26f0338bd4663ec6952a2/components/o-private-foundation/src/scss/o-buttons/main.scss#L233-L265).

The usage of `oPrivateButtonsContentWithThemeOverride` is used in approx six components which will be part of this refactor.

How are you feeling about making this deletion? I feel like it's going to add a few more days work onto this so might not be the best value to add to this PR. At the very top-level. The colors used to generate at the top-level are from o3.

**Review by notlee — APPROVED** [review]
